### PR TITLE
Tensor: Added _eval_simplify method for class ImmutableDenseNDimArray.

### LINKED ISSUE #18361: simplify not working on Array (but working on sympy expressions)
As far as I remember, simplify could operate on Arrays previously. Kindly see the snippet below :
```
>>> from sympy.abc import x
>>> from sympy import sin, cos, simplify, Array
>>> import sympy
>>> sympy.__version__
'1.5'
>>> A = Array([sin(2*x) - 2*sin(x)*cos(x), 0])
>>> simplify(A)
[-2*sin(x)*cos(x) + sin(2*x), 0]
>>> simplify(sin(2*x) - 2*sin(x)*cos(x))
0

```

Now it is not working on Array, but working on same expressions when each element on Array is simplified individually.

### PR DESCRIPTION
**Pull request by iamabhishek0**
#### References to other Issues or PRs

Fixes #18361 

#### Brief description of what is fixed or changed
--> _evalf_simplify simplifies each element of the array.
--> Added Tests

#### Other comments
```
>>> from sympy.abc import x, y
>>> from sympy import sin, cos, simplify, Array
>>> B = Array([sin(x)**2 + cos(x)**2, 0])
>>> simplify(B)
[1, 0]
>>> C = Array([(x + x**2)/(x*sin(y)**2 + x*cos(y)**2), 2*sin(x)*cos(x)])
>>> simplify(C)
[x + 1, sin(2*x)]
```

#### Release Notes

* tensor
  * Added _eval_simplify method for class ImmutableDenseNDimArray

### CODE REVIEW

**Comment by sylee957** [comment]
It should not return in a python native object.
You should better use Array.applyfunc(simplify)

**Comment by iamabhishek0** [comment]
Thanks @sylee957 for the suggestion.
Have updated the PR.

File: sympy/tensor/array/dense_ndim_array.py [hunk 1]
```diff
@@ -150,6 +150,8 @@ def __setitem__(self, index, value):
     def as_mutable(self):
         return MutableDenseNDimArray(self)
 
+    def _eval_simplify(self, **kwargs):
+        return self.applyfunc(simplify)
```
Comment (Upabjojr) [inline sympy/tensor/array/dense_ndim_array.py]: I would rather recommend `self.applyfunc(lambda x: x._eval_simplify(**kwargs))`... `simplify` is the method available to end-users, `_eval_simplify` is for the internal code (`simplify` does a lot of parsing and error checking before calling `._eval_simplify`... once you have entered `_eval_simplify` it doesn't make any sense to call `simplify`).

Comment (iamabhishek0) [reply sympy/tensor/array/dense_ndim_array.py]: When returning `self.applyfunc(lambda x: x._eval_simplify(**kwargs))` it raises AttributeError.
```>>> from sympy.abc import x
>>> from sympy import sin, cos, simplify, Array
>>> A = Array([sin(2*x) - 2*sin(x)*cos(x), 0])
>>> simplify(A)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/uchiha/Desktop/sympy/sympy/simplify/simplify.py", line 559, in simplify
    return _eval_simplify(**kwargs)
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/dense_ndim_array.py", line 154, in _eval_simplify
    return self.applyfunc(lambda x: x._eval_simplify(**kwargs))
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/ndim_array.py", line 292, in applyfunc
    return type(self)(map(f, Flatten(self)), self.shape)
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/dense_ndim_array.py", line 129, in __new__
    return cls._new(iterable, shape, **kwargs)
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/dense_ndim_array.py", line 138, in _new
    flat_list = flatten(flat_list)
  File "/home/uchiha/Desktop/sympy/sympy/utilities/iterables.py", line 75, in flatten
    for el in iterable:
  File "/home/uchiha/Desktop/sympy/sympy/tensor/array/dense_ndim_array.py", line 154, in <lambda>
    return self.applyfunc(lambda x: x._eval_simplify(**kwargs))
AttributeError: 'Add' object has no attribute '_eval_simplify'
```

Comment (iamabhishek0) [reply sympy/tensor/array/dense_ndim_array.py]: Ping @Upabjojr

Comment (sylee957) [reply sympy/tensor/array/dense_ndim_array.py]: @Upabjojr I may also clarify that not every sympy object has `_eval_simplify`.

Comment (mcpl-sympy) [reply sympy/tensor/array/dense_ndim_array.py]: @Upabjojr  By looking at the code, it seems that `simplify` checks only one condition (that if `expr` being scalar zero) before calling `_eval_simplify` method of expression.

**Comment by mcpl-sympy** [comment]
> It should not return in a python native object.
> You should better use Array.applyfunc(simplify)

@sylee957 Could you give more explanation of why the previous one was unsuitable?

**Comment by sylee957** [comment]
SymPy expects some functions to give 'sympified' outputs.
Of course, not the ones like in `ntheory` module, but for `simplify`, it should be.

**Comment by iamabhishek0** [comment]
@sylee957 If that's the case, then I guess `return self.applyfunc(simplify)` should be used here.

**Comment by sylee957** [comment]
Well, he had done exactly as `return self.applyfunc(simplify)` and that's what I had actually suggested.

**Pull request merged**